library/src/factory.cxx.in

```diff
@@ -60,6 +60,43 @@ reader* factory::getReader(const std::string& fileName)
   return bestReader;
 }
 
+//----------------------------------------------------------------------------
+bool factory::setReaderOption(const std::string& name, const std::string& value)
+{
+  bool found = false;
+  for (auto plugin : this->Plugins)
+  {
+    for (auto reader : plugin->getReaders())
+    {
+      if (reader->setReaderOption(name, value))
+      {
+        found = true;
+        break;
+      }
+    }
+    if (found)
+    {
+      break;
+    }
+  }
+  return found;
```

Comment:
Did I miss something, or is it equivalent?
```suggestion
  for (auto plugin : this->Plugins)
  {
    for (auto reader : plugin->getReaders())
    {
      if (reader->setReaderOption(name, value))
      {
        return true;
      }
    }
  }
  return false;
```

Reply:
You may also want to use reference when possible while looping on plugins/readers